File: Swap_Position.py
def swap(self, position1, position2):
    if self.is_empty() or position1 < 0 or position2 < 0 or position1 > self.rear - self.front or position2 > self.rear - self.front:
        print("Invalid positions for swap.")
        return

    index1 = self.front + position1
    index2 = self.front + position2
    self.array[index1], self.array[index2] = self.array[index2], self.array[index1]

File: Swap_Position.c
void swap(struct Queue *queue, int position1, int position2) {
    if (isEmpty(queue) || position1 < 0 || position2 < 0 || position1 > queue->rear - queue->front || position2 > queue->rear - queue->front) {
        printf("Invalid positions for swap.\n");
        return;
    }

    int index1 = queue->front + position1;
    int index2 = queue->front + position2;
    int temp = queue->array[index1];
    queue->array[index1] = queue->array[index2];
    queue->array[index2] = temp;
}
